src/dsmr/reader.h

```diff
@@ -218,10 +221,10 @@ class P1Reader {
     Stream *stream;
     uint8_t req_pin;
     enum class State : uint8_t {
-      DISABLED,
-      WAITING,
-      READING,
-      CHECKSUM,
+      DISABLED_STATE,
+      WAITING_STATE,
+      READING_STATE,
+      CHECKSUM_STATE,
```

Comment:
Ah, right. This should certainly be documented in the commit message, then. Also, perhaps you could file a bug report with the ESP8266 about this, since IMHO it's a bug to define generic macros like these. If they can replace the macro with a global constant with the same value, the compiler can properly apply scoping rules and this will not a problem. We should probably *also* change this in my library to make things work with current versions (though I don't quite like the extra verbosity of these constant names, but I can't think of any better alternatives right now).